addons/crm_iap_enrich/models/crm_lead.py

```diff
@@ -25,16 +26,18 @@ def _compute_show_enrich_button(self):
                 lead.show_enrich_button = True
 
     @api.model
-    def _iap_enrich_leads_cron(self, enrich_hours_delay=24, leads_batch_size=1000):
+    def _iap_enrich_leads_cron(self, enrich_hours_delay=24, batch_size=50):
         timeDelta = self.env.cr.now() - datetime.timedelta(hours=enrich_hours_delay)
         # Get all leads not lost nor won (lost: active = False)
         leads = self.search([
             ('iap_enrich_done', '=', False),
-            ('reveal_id', '=', False),
             '|', ('probability', '<', 100), ('probability', '=', False),
-            ('create_date', '>', timeDelta)
-        ], limit=leads_batch_size)
-        leads.iap_enrich(from_cron=True)
+            ('email_from', '!=', False),
+            ('reveal_id', '=', False),
+            ('create_date', '>', timeDelta),
+            ('active', '=', True),
```

Comment:
Why the explicit `active` check ?

Reply:
At some point, I thought of using the domain for filtered (which does not have active_test). I did not remove it, because it is often better be explicit than implicit.